x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutTrainedModelDefinitionPartAction.java

```diff
@@ -78,6 +91,11 @@ public Request(StreamInput in) throws IOException {
             this.part = in.readVInt();
             this.totalDefinitionLength = in.readVLong();
             this.totalParts = in.readVInt();
+            if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_038)) {
```

Comment:
I _think_ we need this version check right?

Reply:
We need a new version constant (which would currently be `V_8_500_040`). There's a comment in `TransportVersion.java` about how to add a new one.

If `V_8_500_038` were used here then serverless upgrades would fail, as already-deployed nodes in serverless will think they understand everything included in `V_8_500_038`.

Reply:
Thanks Dave! Yeah I forgot to merge upstream so I added 38 but someone had already added it. I'll add a new one 👍